service/src/main/java/bio/terra/drshub/services/AuthService.java

```diff
@@ -184,17 +185,24 @@ private DrsHubAuthorization mapAccessMethodConfigAuthType(
   @VisibleForTesting
   Optional<Authorizations> fetchDrsAuthorizations(
       DrsProvider drsProvider, UriComponents uriComponents) {
-    var drsApi = drsApiFactory.getApiFromUriComponents(uriComponents, drsProvider);
+    var drsApi =
+        Optional.ofNullable(drsApiFactory.getApiFromUriComponents(uriComponents, drsProvider));
     var objectId = uriComponents.getPath();
     try {
-      return Optional.ofNullable(drsApi.optionsObject(objectId));
+      return Optional.ofNullable(drsApi.orElseThrow().optionsObject(objectId));
     } catch (RestClientException ex) {
       log.warn(
           "Failed to get authorizations for {} from OPTIONS endpoint for DRS Provider {}. "
               + "Falling back to configured authorizations",
           objectId,
           drsProvider.getName());
       return Optional.empty();
+    } catch (NoSuchElementException ex) {
+      throw new DrsHubException(
+          String.format(
+              "Failed to initialize DrsApi for provider %s and uri components %s. "
+                  + "You may have passed in a malformed DRS url, or we do not support this provider.",
+              drsProvider.getName(), uriComponents.toUriString()));
```

Comment:
Some tests were passing in malformed drs urls and getting nullpointer exceptions.